### PR TITLE
Add Tiny Integration and E2E Integration tests

### PR DESCRIPTION
**Pull request by aws-navyadhara**
## What is the change? 

Added tests to run in CI/CD and validate changes.

#### Features

| Feature | Test |
|---------|------|
| Continuous Batching | `test_vanilla_inference_tiny`, `test_continuous_batching_inference`, `test_online_integration` |
| Chunked Prefill | `test_chunked_prefill_inference` |
| Prefix Caching (APC) | `test_prefix_caching_inference`, `test_online_integration` |
| Speculative Decoding (EAGLE) | `test_eagle_speculative_decoding`, `test_online_eagle` |
| Dynamic Sampling | `test_dynamic_sampling` |
| Tool Calling | `test_online_tool_calling` |
| Multi-modal | `test_llama4` |
| Edge Cases | `test_min_tokens_trumps_eos`, `test_high_token_budget` , `test_double_on_device_sampling_config` |
| Qwen | `test_qwen` |

#### Models

| Model | Test |
|-------|------|
| TinyLlama/TinyLlama-1.1B-Chat-v1.0 | Edge cases |
| meta-llama/Llama-2-7b-chat-hf | `test_eagle_speculative_decoding` |
| meta-llama/Llama-3.1-8B-Instruct | `test_online_integration`, `test_online_eagle` |
| meta-llama/Llama-3.1-70B | `test_online_tool_calling` |
| meta-llama/Llama-4-Scout-17B-16E-Instruct | `test_llama4` |
| openlm-research/open_llama_3b | `test_openllama3b_batch_variations` |
| openlm-research/open_llama_7b | `test_online_integration` |
| Qwen/Qwen2.5-7B-Instruct | `test_qwen` |
| Qwen/Qwen3-8B | `test_online_integration` |
| yuhuili/EAGLE-llama2-chat-7B | `test_eagle_speculative_decoding` |
| yuhuili/EAGLE-LLaMA3-Instruct-8B | `test_online_eagle` |

### CODE REVIEW

**Comment by elaineyz** [comment]
I would move the `test_double_on_device_sampling_config` test under Edge Cases feature category. Other than that, LGTM

File: test/e2e/online/README.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+├── e2e/
+│   └── online/
+│       ├── configs.py                   # Named presets: model path, TP, batch size, ports, token settings, etc.
+│       ├── online_server_runner.py      # Spins up/tears down vLLM server (VllmServer)
+│       ├── test_online_e2e.py           # Consolidated e2e tests (vanilla, eagle, qwen, tool calling, reasoning)
+│       └── test_online_multi_lora.py    # Multi-LoRA specific e2e test
+├── tiny/
+│   └── test_vanilla_inference_tiny.py   # Tiny/smoke test for PR suite
+└── utils/
+    ├── model_path.py                 # resolve_model_dir(): FSX → SSD (if writable) path resolver
```
Comment (elaineyz) [inline test/e2e/online/README.md]: missing `fsx_utils` layer here

Comment (aws-navyadhara) [reply test/e2e/online/README.md]: Thanks for catching will update the README

File: test/utils/server/tool_chat_template_llama3.1_json.jinja
Comment (aws-satyajith) [inline test/utils/server/tool_chat_template_llama3.1_json.jinja]: This file seems like it's not needed or used anywhere. Can we remove it?

**Review by aws-satyajith — CHANGES_REQUESTED** [review]

**Review by aws-satyajith — APPROVED** [review]

**Review by elaineyz — APPROVED** [review]
